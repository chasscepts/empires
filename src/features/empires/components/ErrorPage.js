import PropTypes from 'prop-types';
import AppHeader from './AppHeader';

const styles = {
  h1: {
    textAlign: 'center',
    marginTop: '40px',
  },
};

export default function ErrorPage({ message }) {
  return (
    <>
      <AppHeader />
      <h1 style={styles.h1}>{message}</h1>
    </>
  );
}

ErrorPage.propTypes = {
  message: PropTypes.string,
};

ErrorPage.defaultProps = {
  message: 'An Error occurred while loading page!',
};
