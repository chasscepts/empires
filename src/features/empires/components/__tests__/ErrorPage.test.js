import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import ErrorPage from '../ErrorPage';

const message = 'Test Error';

it('matches snapshot', () => {
  const tree = renderer
    .create(<ErrorPage message={message} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('AppHeader', () => {
  beforeEach(() => {
    render(<ErrorPage message={message} />);
  });

  it('displays the heading', () => {
    expect(screen.getByText('Error')).toBeInTheDocument();
  });

  it('displays the message', () => {
    expect(screen.getByText(message)).toBeInTheDocument();
  });
});
