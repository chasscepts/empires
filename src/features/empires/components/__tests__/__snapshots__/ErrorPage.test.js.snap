// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrap"
>
  <div>
    <div
      className="headingWrap"
    >
      <svg
        className="svg"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 14H11V9H13M13 18H11V16H13M1 21H23L12 2L1 21Z"
          fill="#c31212"
        />
      </svg>
      <div
        className="heading"
      >
        Error
      </div>
    </div>
    <p
      className="message"
    >
      Test Error
    </p>
  </div>
</div>
`;
